Fix centering window on linux (fixes #122)

dbhowell opened the pull request:
This is a workaround for #122 . On Linux there is an open issue with electron (See electron/electron#3490).

Since the electron has been outstanding for quite a while, I can't see it being fixed soon.

This workaround finds the window that mostly intersects the window and uses those bounds to centre the window.

It isn't perfect, but does a better job than splitting across multiple windows.

bengotow commented:
Hey! This looks great. Could you make two minor tweaks? In electron apps, the operating system is available on `process.platform` (so it'd be `process.platform === "linux"`) Could you change this to use that? I think using `os` is just as good, but just in case someone is grepping for everywhere that we detect linux one day it'd be nice to use the same method.

Also I'm worried that there might be scenarios where `getDisplayMatching` returns `null` instead of a screen. I tried to find the source for it but it looks like it's all the way down in Chromium. Could you make it so that this falls back to calling `center` if it can't find an appropriate display?

Thanks for contributing this!

dbhowell commented:
I've updated the fix to use `process.platform`. Also, I've modified it to fallback to `getPrimaryDisplay` in the case that `getDisplayMatching` returns `null`. That should cover it.

bengotow commented:
Looks great - thank you!
